langchain/src/util/sql_utils.ts

```diff
@@ -177,17 +177,28 @@ export const getTableAndColumnsName = async (
   }
 
   if (appDataSource.options.type === "mssql") {
-    sql =
+    const schema = appDataSource.options?.schema;
+    const sqlWithSchema =
       "SELECT " +
       "TABLE_NAME AS table_name, " +
       "COLUMN_NAME AS column_name, " +
       "DATA_TYPE AS data_type, " +
       "IS_NULLABLE AS is_nullable " +
       "FROM INFORMATION_SCHEMA.COLUMNS " +
+      `WHERE TABLE_SCHEMA = '${schema}'` +
```

Comment:
If this can be undefined, should we do this instead?
```suggestion
      schema ? `WHERE TABLE_SCHEMA = '${schema}'` : "" +
```

Reply:
This way we can just drop `sqlWithoutSchema`, see my below comment

Reply:
When there is no schema in the dataSource config, it is throwing syntax error ⇒ Incorrect syntax near the keyword. I think this is because when schema is not present in the configuration, WHERE TABLE_SCHEMA = "" denotes an empty string as schema name and “” is not a valid schema name.

Reply:
However, still removed the unnecessary code as you pointed out and made a new commit which is working as expected